src/array_value.cpp

```diff
@@ -951,11 +951,32 @@ void ArrayValue::push_splat(Env *env, ValuePtr val) {
     }
 }
 
-ValuePtr ArrayValue::pop(Env *env) {
+ValuePtr ArrayValue::pop(Env *env, ValuePtr count) {
     this->assert_not_frozen(env);
+
+    if (count) {
+        auto to_int = SymbolValue::intern("to_int");
+
+        if (!count->is_integer() && count->respond_to(env, to_int))
+            count = count.send(env, to_int);
+
+        count->assert_type(env, ValueType::Integer, "Integer");
+        auto c = count->as_integer()->to_nat_int_t();
+
+        if (c < 0)
+            env->raise("ArgumentError", "negative array size");
+
+        if (c > (nat_int_t)size())
+            c = (nat_int_t)size();
```

Comment:
Any time I coerce a `size_t` to `nat_int_t` I add an `assert(size() <= NAT_INT_MAX);` just prior. I suppose that's overkill since an array of that size would likely consume all the memory in my machine. But still, seems like a good habit to make. Thoughts?

Reply:
Makes sense, will fix it if I start cleaning up Array or when I implement the next method :) Thanks!